app/services/retrieve_pipeline_viz_graph_data_service.rb

```diff
@@ -83,6 +88,12 @@ def create_edges
     return edges
   end
 
+  def step_status
+    @pipeline_run.pipeline_run_stages.map do |prs|
+      JSON.parse(PipelineOutputsHelper.get_s3_file(prs.step_status_file_path) || "{}")
+    end
```

Comment:
Is it guaranteed to be correct JSON?
Maybe we should rescue parser error, just in case

Reply:
It should be correct unless the file is written incorrectly, but I'll add the rescue parser error too 👍